dlls/ff/ff_player.cpp

```diff
@@ -8034,9 +8034,13 @@ void CFFPlayer::UpdateCamera( bool bUnassigned )
 // purpose: update our vector of recent attackers for kill assists
 void CFFPlayer::AddRecentAttacker( const CTakeDamageInfo &dmgInfo )
 {
-	CFFPlayer *pAttacker = ToFFPlayer( dmgInfo.GetAttacker() );
+	// make sure to get the effective scorer incase dmg is from
+	// a buildable - det, sg, disp, etc
+	CFFPlayer *pAttacker = ToFFPlayer(dynamic_cast<CMultiplayRules *>(g_pGameRules)->GetDeathScorer( dmgInfo.GetAttacker(), dmgInfo.GetInflictor() ));

```

Comment:
actually,  I was surprised, downcast is needed here
`error C2039: 'GetDeathScorer' : is not a member of 'CGameRules'`